rclcpp/include/rclcpp/create_subscription.hpp

```diff
@@ -173,6 +174,57 @@ template<
   typename SubscriptionT = rclcpp::Subscription<MessageT, AllocatorT>,
   typename MessageMemoryStrategyT = typename SubscriptionT::MessageMemoryStrategyType,
   typename NodeT>
+[[deprecated("use another overloaded method create_subscription instead")]]
```

Comment:
we should mention that the purpose of the deprecation is to have a separate argument for the callback group

Reply:
I think the message in the deprecated can be either the reason or suggestion, what about adding the reason  [here](https://github.com/ros2/rclcpp/blob/126eeee3aaecb94e088fff24ee24a3743f898755/rclcpp/include/rclcpp/create_subscription.hpp#L146-L153)?

Reply:
I'll follow your suggestion, what about
```suggestion
[[deprecated(
  "To Keep consistency with other cases(Client, Service, Timer, etc) not hold callback_group,\n" \
  "the callback_group in the SubscriptionOptionsBase will be removed in the future since the\n" \
  "SubscriptionOptions is held in the Subscription and then the callback_group is depend\n" \
  "on the lifetime of Subscription.\n" \
  "Use\n" \
  "  template<\n" \
  "    typename MessageT,\n" \
  "    typename CallbackT,\n" \
  "    typename AllocatorT = std::allocator<void>,\n" \
  "    typename SubscriptionT = rclcpp::Subscription<MessageT, AllocatorT>,\n" \
  "    typename MessageMemoryStrategyT = typename SubscriptionT::MessageMemoryStrategyType\n" \
  "    typename NodeT>\n" \
  "  typename std::shared_ptr<SubscriptionT>\n" \
  "  create_subscription(\n" \
  "    NodeT & node,\n" \
  "    const std::string & topic_name,\n" \
  "    const rclcpp::QoS & qos,\n" \
  "    CallbackT && callback,\n" \
  "    const SubscriptionOptionsWithAllocator<AllocatorT> & options =(\n" \
  "      SubscriptionOptionsWithAllocator<AllocatorT>()\n" \
  "    ),\n" \
  "    rclcpp::CallbackGroup::SharedPtr group = nullptr,\n" \
  "    typename MessageMemoryStrategyT::SharedPtr msg_mem_strat = (\n" \
  "      MessageMemoryStrategyT::create_default()\n" \
  "    )\n" \
  "  );\n" \
  "instead"
)]]
```